### PR TITLE
K8s policies

### PR DESCRIPTION
**Pull request by gabrielscribe**
Worked on K8s policies, using namespace and pod attestation

### CODE REVIEW

File: discovery/README.md [hunk 1]
```diff
@@ -5,52 +5,52 @@ This README provides an overview of the set of policy rules that utilize discove
...
-    2. [Token](#token)
-        1. [Unused Tokens](#unused-tokens)
-        2. [Shortly Expiring Tokens](#shortly-expiring-tokens)
-        3. [Tokens with too long lifetime](#tokens-with-too-long-lifetime)
-        4. [Forbid Specific Scopes](#forbid-specific-scopes)
-        5. [Allowed Scopes](#allowed-scopes)
-    3. [Projects](#projects)
-        1. [Projects Visibility](#projects-visibility)
-        2. [Inactive Projects](#inactive-projects)
+   1. [User](#user)
```
Comment (cppvik) [inline discovery/README.md:8]: please remove unnecessary changes in the tabs

File: v1/k8s/secrets-expiration.rego [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+			"is_expired": true,
+		} 
+
+
+	}
+}
+
+is_expired(secret) {
+
+	# Must implement a way to check if a secret is expired based off how the object is formed
```
Comment (cppvik) [inline v1/k8s/secrets-expiration.rego:74]: Can you finish this one please?

File: v1/k8s/white-listed-namespace.rego [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+
+
+# j is now a list in order to make sure duplications are not lost
+
+violations = j {
+	j := [r |
+
+		projects := object.remove(input.evidence.predicate.content, {"metadata"})
```
Comment (cppvik) [inline v1/k8s/white-listed-namespace.rego:51]: We shouldn't modify input in any way

File: v1/k8s/use-of-a-specific-secret.rego [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+			"is_expired": true,
+		} 
+
+
+	}
+}
+
+matches(secret) {
+
+	# Must implement a way to check if a secret is expired based off how the object is formed
```
Comment (cppvik) [inline v1/k8s/use-of-a-specific-secret.rego:74]: Can you implement this one, please?

File: v1/k8s/long-running-images.rego [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+			"id": pod.id,
+			"query_id": pod.query_id,
+			"image": image,
+			"last_started": container_info.lastState[_].startedAt
+        }
+	]
+}
+
+
+# Only checks if it terminated, doesnt check if it ran for a long time.
```
Comment (cppvik) [inline v1/k8s/long-running-images.rego:60]: Why so?

File: v1/k8s/white-listed-pod.rego [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+		pod := project.pod
+		
+		not is_valid(pod)
+
+		r := {
+			"scribe_type": pod.scribe_type,
+            "name": pod.name,
+			"id": pod.id,
+			"query_id": pod.query_id,
+			"white_listed_pod": white_listed_pod,
```
Comment (cppvik) [inline v1/k8s/white-listed-pod.rego:63]:
```suggestion
```

File: v1/k8s/white-listed-namespace.rego [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+		namespace := project.namespace
+		
+		not is_valid(namespace)
+
+		r := {
+			"scribe_type": namespace.scribe_type,
+            "name": namespace.name,
+			"id": namespace.id,
+			"query_id": namespace.query_id,
+			"white_listed_namespace": white_listed_namespace,
```
Comment (cppvik) [inline v1/k8s/white-listed-namespace.rego:63]:
```suggestion
```

File: v1/k8s/use-of-a-specific-secret.yaml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+evidence:
+  signed: false
+  format-type: generic
+  target_type: data
+  predicate_type: http://scribesecurity.com/evidence/discovery/v0.1
+  labels:
+    - "asset_type=pod"
+
+# Implementation not complete
```
Comment (cppvik) [inline v1/k8s/use-of-a-specific-secret.yaml:16]: TODO: remove

File: v1/k8s/use-of-a-specific-secret.rego (hunk 4 above)
Comment (gabrielscribe) [reply v1/k8s/use-of-a-specific-secret.rego:74]: None of the attestation i received had any secrets, so i wasnt able to implement a way to check the secrets

**Comment by gabrielscribe** [comment]
I cannot implement the secret policies since the attestations do not have secrets, thus i don't know how the secret object looks like

**Review by cppvik — APPROVED** [review]
